cetmix_tower_server/models/cx_tower_server.py

```diff
@@ -762,71 +684,101 @@ def test_ssh_connection(
 
         return test_result
 
-    def _render_command(self, command, path=None):
-        """Renders command code for selected command for current server
+    def _get_connection_test_command(self):
+        """Get command used to test SSH connection
 
-        Args:
-            command (cx.tower.command): Command to render
-            path (Char): Path where to run the command.
-                Provide in case you need to override default command path
+        Returns:
+            Char: SSH command
+        """
+        command = "uname -a"
+        return command
+
+    def _get_ssh_password(self):
+        """Get ssh password
+        This function prepares and returns ssh password for the ssh connection
+        Override this function to implement own password algorithms
 
         Returns:
-            dict: rendered values
-                {
-                    "rendered_code": rendered command code,
-                    "rendered_path": rendered command path
-                }
+            Char: password ready to be used for connection parameters
         """
         self.ensure_one()
+        password = self.ssh_password
+        return password
 
-        variables = []
+    def _get_ssh_key(self):
+        """Get SSH key
+        Get private key for an SSH connection
 
-        # Get variables from code
-        if command.code:
-            variables_extracted = command.get_variables_from_code(command.code)
-            for ve in variables_extracted:
-                if ve not in variables:
-                    variables.append(ve)
+        Returns:
+            Char: SSH private key
+        """
+        self.ensure_one()
+        # To ensure that key will be read
+        # regardless of access rights
+        self = self.sudo()
+        if self.ssh_key_id:
+            # Use context key to read secret value
+            ssh_key = (
+                self.ssh_key_id.sudo()
+                .with_context(show_secret_value=True)
+                .read(["secret_value"])[0]["secret_value"]
+            )
+        else:
+            ssh_key = None
+        return ssh_key
 
-        # Get variables from path
-        path = path if path else command.path
-        if path:
-            variables_extracted = command.get_variables_from_code(path)
-            for ve in variables_extracted:
-                if ve not in variables:
-                    variables.append(ve)
+    @after_commit
```

Comment:
perhaps we can also rename this decorator to a more understandable one e.g.: 

`@with_ssh_cleanup`

what do you think?

Reply:
Sounds like a good idea!